# Part B - Linear Regression

# Reading and Pre-processing Data

In [1]:
import pandas as pd
dataframe=pd.read_csv('FW_Veg_Rem_Combined.csv')
df=dataframe.copy()
df.keys()

Index(['Unnamed: 0', 'Unnamed: 0.1', 'fire_name', 'fire_size',
       'fire_size_class', 'stat_cause_descr', 'latitude', 'longitude', 'state',
       'disc_clean_date', 'cont_clean_date', 'discovery_month',
       'disc_date_final', 'cont_date_final', 'putout_time', 'disc_date_pre',
       'disc_pre_year', 'disc_pre_month', 'wstation_usaf', 'dstation_m',
       'wstation_wban', 'wstation_byear', 'wstation_eyear', 'Vegetation',
       'fire_mag', 'weather_file', 'Temp_pre_30', 'Temp_pre_15', 'Temp_pre_7',
       'Temp_cont', 'Wind_pre_30', 'Wind_pre_15', 'Wind_pre_7', 'Wind_cont',
       'Hum_pre_30', 'Hum_pre_15', 'Hum_pre_7', 'Hum_cont', 'Prec_pre_30',
       'Prec_pre_15', 'Prec_pre_7', 'Prec_cont', 'remoteness'],
      dtype='object')

In [2]:
df.head()

Unnamed: 0  Unnamed: 0.1 fire_name  fire_size fire_size_class  \
0           0             0       NaN       10.0               C   
1           1             1       NaN        3.0               B   
2           2             2       NaN       60.0               C   
3           3             3    WNA  1        1.0               B   
4           4             4       NaN        2.0               B   

    stat_cause_descr   latitude   longitude state disc_clean_date  ...  \
0  Missing/Undefined  18.105072  -66.753044    PR       2/11/2007  ...   
1              Arson  35.038330  -87.610000    TN      12/11/2006  ...   
2              Arson  34.947800  -88.722500    MS       2/29/2004  ...   
3     Debris Burning  39.641400 -119.308300    NV        6/6/2005  ...   
4      Miscellaneous  30.700600  -90.591400    LA       9/22/1999  ...   

  Wind_cont Hum_pre_30 Hum_pre_15  Hum_pre_7   Hum_cont Prec_pre_30  \
0  3.250413  78.216590  76.793750  76.381579  78.724370         0.0   
1  2.122320  70.840000  65.858911  55.505882  81.682678        59.8   
2  3.369050  75.531629  75.868613  76.812834  65.063800       168.8   
3  0.000000  44.778429  37.140811  35.353846   0.000000        10.4   
4 -1.000000  -1.000000  -1.000000  -1.000000  -1.000000        -1.0   

   Prec_pre_15 Prec_pre_7 Prec_cont  remoteness  
0          0.0        0.0       0.0    0.017923  
1          8.4        0.0      86.8    0.184355  
2         42.2       18.1     124.5    0.194544  
3          7.2        0.0       0.0    0.487447  
4         -1.0       -1.0      -1.0    0.214633  

[5 rows x 43 columns]

In [3]:
import numpy as np
df.dropna(axis='rows', how='any', inplace=True)
df.head()

Unnamed: 0  Unnamed: 0.1         fire_name  fire_size fire_size_class  \
3            3             3            WNA  1       1.00               B   
9            9             9             CRAZY       1.00               B   
26          26            26        STO3573155       2.00               B   
31          31            31       AXE RONDEAU       1.20               B   
35          35            35  FY2001-EARLY-021      30.18               C   

   stat_cause_descr  latitude  longitude state disc_clean_date  ... Wind_cont  \
3    Debris Burning   39.6414  -119.3083    NV        6/6/2005  ...       0.0   
9     Miscellaneous   48.8394   -99.7185    ND       4/19/2004  ...       0.0   
26   Debris Burning   35.7189   -92.3350    AR       1/19/2014  ...       0.0   
31   Debris Burning   48.8330   -99.7836    ND        4/7/2007  ...       0.0   
35   Debris Burning   31.2590   -84.8956    GA      10/19/2000  ...       0.0   

   Hum_pre_30 Hum_pre_15  Hum_pre_7 Hum_cont Prec_pre_30  Prec_pre_15  \
3   44.778429  37.140811  35.353846      0.0        10.4          7.2   
9   64.606509  55.943038  54.337838      0.0        12.3          1.8   
26  76.091398   0.000000   0.000000      0.0         0.0          0.0   
31  77.575012  75.963981  71.173116      0.0        21.2         21.2   
35  67.551783  61.733788  60.328571      0.0         0.0          0.0   

   Prec_pre_7 Prec_cont  remoteness  
3         0.0       0.0    0.487447  
9         0.0       0.0    0.291683  
26        0.0       0.0    0.227270  
31        9.2       0.0    0.292303  
35        0.0       0.0    0.162416  

[5 rows x 43 columns]

In [4]:
from sklearn.model_selection import train_test_split
training_set, testing_set = train_test_split(df, test_size=0.30, random_state=590)
print(training_set.info())
print(testing_set.info())

<class 'pandas.core.frame.DataFrame'>
Int64Index: 12341 entries, 12081 to 30737
Data columns (total 43 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Unnamed: 0        12341 non-null  int64  
 1   Unnamed: 0.1      12341 non-null  int64  
 2   fire_name         12341 non-null  object 
 3   fire_size         12341 non-null  float64
 4   fire_size_class   12341 non-null  object 
 5   stat_cause_descr  12341 non-null  object 
 6   latitude          12341 non-null  float64
 7   longitude         12341 non-null  float64
 8   state             12341 non-null  object 
 9   disc_clean_date   12341 non-null  object 
 10  cont_clean_date   12341 non-null  object 
 11  discovery_month   12341 non-null  object 
 12  disc_date_final   12341 non-null  object 
 13  cont_date_final   12341 non-null  object 
 14  putout_time       12341 non-null  object 
 15  disc_date_pre     12341 non-null  object 
 16  disc_pre_year     12341 non-null  in

In [5]:
training_set.head()

Unnamed: 0  Unnamed: 0.1          fire_name  fire_size fire_size_class  \
12081       12081         12081  EMMONS CREEK FIRE      36.83               C   
51799       51799         51799            GRIZZLY    5832.20               G   
31604       31604         31604            MM9 I84       0.60               B   
10058       10058         10058           CRABTREE       0.60               B   
38629       38629         38629     GFX ASSIST 578       3.00               B   

        stat_cause_descr   latitude   longitude state disc_clean_date  ...  \
12081              Arson  44.300240  -89.250310    WI       4/13/2015  ...   
51799          Lightning  42.228889 -120.552222    OR       7/12/2002  ...   
31604      Equipment Use  43.913620 -116.823880    ID       7/15/2014  ...   
10058          Powerline  36.429500  -84.943650    TN      10/13/2007  ...   
38629  Missing/Undefined  39.480300 -108.007200    CO       9/26/2009  ...   

      Wind_cont Hum_pre_30 Hum_pre_15  Hum_pre_7   Hum_cont Prec_pre_30  \
12081  0.000000  57.537106  62.994424  70.312000   0.000000        51.7   
51799  3.217777  44.153846  42.087891  36.876333  38.952563        35.6   
31604  0.000000  36.937415  29.601648  29.970238   0.000000         6.9   
10058  1.545783  69.148684  66.460894  70.729885  72.928571        30.2   
38629 -1.000000  -1.000000  -1.000000  -1.000000  -1.000000        -1.0   

       Prec_pre_15 Prec_pre_7 Prec_cont  remoteness  
12081         45.9       40.2       0.0    0.195039  
51799         35.6        0.0      24.9    0.135625  
31604          0.0        0.0       0.0    0.460576  
10058         21.3       14.2       1.0    0.159415  
38629         -1.0       -1.0      -1.0    0.374556  

[5 rows x 43 columns]

In [6]:
testing_set.head()

Unnamed: 0  Unnamed: 0.1      fire_name  fire_size fire_size_class  \
51856       51856         51856     YETNA RIVR   152962.0               G   
44701       44701         44701     CHOKCHERRY        1.0               B   
38989       38989         38989        CLIPPER        3.0               B   
662           662           662            CIG        1.0               B   
24859       24859         24859  F2001500361.0       21.0               C   

      stat_cause_descr   latitude   longitude state disc_clean_date  ...  \
51856        Lightning  62.844300 -158.351000    AK       7/18/2002  ...   
44701   Debris Burning  48.123400 -105.177800    MT        7/3/1995  ...   
38989        Lightning  44.894700 -105.929700    WY       7/17/2000  ...   
662      Miscellaneous  31.916700 -111.849000    AZ       10/1/2010  ...   
24859        Fireworks  37.804456  -81.733825    WV       12/5/2001  ...   

      Wind_cont Hum_pre_30 Hum_pre_15  Hum_pre_7   Hum_cont Prec_pre_30  \
51856  1.708183  45.052997  48.594937  45.364780  45.543489         2.6   
44701  0.000000  57.840909  55.465116  56.315789   0.000000         0.0   
38989 -1.000000  -1.000000  -1.000000  -1.000000  -1.000000        -1.0   
662    0.000000  30.279070  29.214660  25.252747   0.000000         0.0   
24859  0.000000  64.784676  66.656676  65.092391   0.000000        18.9   

       Prec_pre_15 Prec_pre_7 Prec_cont  remoteness  
51856          2.6        0.0       8.1    0.726307  
44701          0.0        0.0       0.0    0.344231  
38989         -1.0       -1.0      -1.0    0.352253  
662            0.0        0.0       0.0    0.416993  
24859         18.9        4.8       0.0    0.129965  

[5 rows x 43 columns]

## 2. Features Selected

I have used the following features for the model:

X = "Temp_pre_7","Wind_pre_7","Hum_pre_7","Prec_pre_7"

Y = "fire_size"

## Reasons

* These values are all float values, hence, computation will be faster (converting String to int or float are time and cost consuming).
* For the purpose of my project: I am planning on predicting the magnitude or size of the fire based on the weather conditions like humidity, wind, temperature and precipitation. The raw data recorded in the dataset that I am using contains these values for the current day of fire, 7 days before the fire, 15 days before the fire and 30 days before the fire.
* Obviously, I cannot use all the feature as it will be overfitting the values to the model. Hence, I am choosing only 4 features, weather conditions from 7 days before the fire.

## 3. Linear Regression


Trainging Set

In [10]:
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error
rege = LinearRegression()
X = training_set[["Temp_pre_7","Wind_pre_7","Hum_pre_7","Prec_pre_7"]]
Y = training_set["fire_size"]
print(rege.fit(X, Y))
print('R2 value is : ',rege.score(X,Y))
y_predict = rege.predict(X)
mean_square = mean_squared_error(Y, y_predict)
print("Mean Squared Error is : ",  mean_square)

LinearRegression()
R2 value is :  0.008392051186792315
Mean Squared Error is :  699982627.9909966


Testing Set

In [8]:
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error
rege = LinearRegression()
X = testing_set[["Temp_pre_7","Wind_pre_7","Hum_pre_7","Prec_pre_7"]]
Y = testing_set["fire_size"]
print(rege.fit(X, Y))
print('R2 value is : ',rege.score(X,Y))
y_predict = rege.predict(X)
mean_square = mean_squared_error(Y, y_predict)
print("Mean Squared Error is : ",  mean_square)

LinearRegression()
R2 value is :  0.010545398726775512
Mean Squared Error is :  471437993.5828763


# Results

Here, I am using R2 value and Mean Square Error as a form of evaluation metrix. The following are the values that I have gotten when I ran Linear Regression for both training and testing sets.

#### Training Set Values:

1. R2 value is :  0.008392051186792315
1. Mean Squared Error is :  699982627.9909966

#### Testing Set Values:

1. R2 value is :  0.010545398726775512
1. Mean Squared Error is :  471437993.5828763